main.py

```diff
@@ -64,6 +65,20 @@ def change_read():
     new_read = ui.get_read_value()     
     book.read = new_read 
     book.save()
+
+
+def delete_book():
+    try:
+        book_id = ui.get_book_id()
+        book = store.get_book_by_id(book_id)
+
+        confirm_deletion = ui.confirm_deletion(book)
+        if confirm_deletion == False:
+            return
+
+        book.delete()
+    except UnboundLocalError:
```

Comment:
UnboundLocalError is one that you very rarely need to catch. If you get this error in your program, it's almost always a coding error, and it's the same error that was fixed in PR #11.   Incorporating changes from #11 you would try to find the book, test if the book is None, if so, it doesn't exist, otherwise it does exist and you can delete it.